hooks/taskcreator.js

```diff
@@ -0,0 +1,93 @@
+var assert      = require('assert');
+var slugid      = require('slugid');
+var taskcluster = require('taskcluster-client');
+var debug       = require('debug')('hooks:taskcreator');
+var _           = require('lodash');
+
+class TaskCreator {
+  /** Create a TaskCreator instance.
+   *
+   * options:
+   * {
+   *   credentials:   // TaskCluster credentials
+   * }
+   * */
+  constructor(options) {
+    assert(options, "options must be given");
+    assert(options.credentials instanceof Object,
+        "Expected credentials");
+    this.credentials = options.credentials;
+  }
+
+  taskForHook(hook) {
```

Comment:
This needs to take a date object called `created`. You can give it a default value of `new Date()`

And all timestamps has to relative to `created`, example:
```js
taskForHook(hook, created = new Date()) {
  assert(created instanceof Date);
  task.created = created.toJSON();
  task.deadline = taskcluster.fromNowJSON(hook.deadline, created);
  ...
}
```

Then use the nextSchedule timestamp as `created` when creating for scheduled task.. Otherwise you're task definition won't be the same, and the idempotent operation will fail if you crash before saving state...